tests/test_router.py

```diff
@@ -53,6 +55,45 @@ def test_other(self):
         from django.contrib.auth import get_user_model
         self.assertEqual(self.sut.db_for_read(model=get_user_model()), None)
 
+    def test_router_gets_filter_hints(self):
+        TestModel.objects.create(user_pk=self.user.pk)
+
+        lookups_to_find = {'exact_lookups': {'user_pk': self.user.pk}}
+
+        with patch.object(ShardedRouter, 'db_for_read') as read_route_function:
+            read_route_function.return_value = 'app_shard_001'
+            test_models = list(TestModel.objects.filter(user_pk=self.user.pk))
+            self.assertIn(lookups_to_find, read_route_function.call_args)
+
+    def test_queryset_router_filter_returns_existing_objects(self):
+        # Cant for sure know what order tests ran in, lets make sure the delete function works in this test
+        TestModel.objects.filter(user_pk=self.user.pk).delete()
+        for i in range(1, 11):
+            test_model_obj = TestModel.objects.create(user_pk=self.user.pk, random_string="%s" % i)
+            self.assertIn(test_model_obj._state.db, ['app_shard_001', 'app_shard_002'])
+
+        test_models = TestModel.objects.filter(user_pk=self.user.pk)
+        self.assertEqual(len(test_models), 10)
```

Comment:
hmmm, then write an equivalent test for count, as I'd rather not take the chance that count does break unnoticed. Aggregates should still work, if you have any way to create that scenario where it does break, I'd love to take a crack at it.